EMS/EMS/SearchEngine.cpp

```diff
@@ -2,31 +2,78 @@
 
 list<Employee*> SearchEngine::search(list<Employee>& db, int employeeNum) const
 {
+	list<Employee*> result;
 
-	return list<Employee*>();
+	for (auto& item : db) {
+		if (item.employeeNum == employeeNum)
+			result.emplace_back(&item);
+	}
+	return result;
 }
 
 list<Employee*> SearchEngine::search(list<Employee>& db, Name& name) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (name.first != "" && name.first != item.name.first) continue;
+		if (name.last != "" && name.last != item.name.last) continue;
+
+		result.emplace_back(&item);
+	}
+
+	return result;
 }
 
-list<Employee*> SearchEngine::search(list<Employee>& db, CL& cl) const
+list<Employee*> SearchEngine::search(list<Employee>& db, CL cl) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (item.cl == cl)
+			result.emplace_back(&item);
```

Comment:
들여쓰기를 줄이기 위해 제안드립니다.
```suggestion
		if (item.cl != cl) continue;
		result.emplace_back(&item);
```